### ISSUE TITLE
sim:mount Failing self test

### ISSUE THREAD

**Issue by btashton** [apache/incubator-nuttx#2089]
`sim:mount` configuration is reporting multiple failures on self test.

```
❯ ./nuttx 
mount_main: mounting vfat filesystem at target=/mnt/fs with source=/dev/ram0
mount_main: mount() returned 0
show_statfs: Try statfs(/mnt)
show_statfs: statfs(/mnt) succeeded
	FS Type           : 9fa0
	Block size        : 0
	Number of blocks  : 0
	Free blocks       : 0
	Free user blocks  : 0
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 32
show_statfs: Try statfs(/mnt/fs)
show_statfs: statfs(/mnt/fs) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1971
	Free user blocks  : 1971
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
dev/
  console
  null
  ram0
mnt/
  fs/
    TESTDIR/
      TESTFILE.TXT
succeed_stat: Try stat(/mnt/fs/TestDir/TestFile.txt)
succeed_stat: stat(/mnt/fs/TestDir/TestFile.txt) succeeded
/mnt/fs/TestDir/TestFile.txt stat:
	mode        : 000081b6
	type        : File
	size        : 15 (bytes)
	block size  : 512 (bytes)
	size        : 1 (blocks)
	access time : 0
	modify time : 0
	change time : 0
show_statfs: Try statfs(/mnt/fs/TestDir/TestFile.txt)
show_statfs: statfs(/mnt/fs/TestDir/TestFile.txt) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1971
	Free user blocks  : 1971
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
read_test_file: opening /mnt/fs/TestDir/TestFile.txt for reading
read_test_file: Read "This is a test
" from /mnt/fs/TestDir/TestFile.txt
fail_stat: Try stat(/mnt/fs/TestDir/WrTest1.txt)
write_test_file: opening /mnt/fs/TestDir/WrTest1.txt for writing
write_test_file: wrote 20 bytes to /mnt/fs/TestDir/WrTest1.txt
dev/
  console
  null
  ram0
mnt/
  fs/
    TESTDIR/
      TESTFILE.TXT
      WRTEST1.TXT
succeed_stat: Try stat(/mnt/fs/TestDir/WrTest1.txt)
succeed_stat: stat(/mnt/fs/TestDir/WrTest1.txt) succeeded
/mnt/fs/TestDir/WrTest1.txt stat:
	mode        : 000081b6
	type        : File
	size        : 20 (bytes)
	block size  : 512 (bytes)
	size        : 1 (blocks)
	access time : 0
	modify time : 0
	change time : 0
show_statfs: Try statfs(/mnt/fs/TestDir/WrTest1.txt)
show_statfs: statfs(/mnt/fs/TestDir/WrTest1.txt) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1970
	Free user blocks  : 1970
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
read_test_file: opening /mnt/fs/TestDir/WrTest1.txt for reading
read_test_file: Read "This is a write test" from /mnt/fs/TestDir/WrTest1.txt
fail_rmdir: Try rmdir(/mnt/fs/TestDir/TestFile.txt)
fail_rmdir: Try rmdir(/mnt/fs/TestDir)
fail_unlink: Try unlink(/mnt/fs/TestDir)
succeed_unlink: Try unlink(/mnt/fs/TestDir/TestFile.txt)
fail_stat: Try stat(/mnt/fs/TestDir/TestFile.txt)
dev/
  console
  null
  ram0
mnt/
  fs/
    TESTDIR/
      WRTEST1.TXT
fail_read_open: Try open(/mnt/fs/TestDir/TestFile.txt) for reading
fail_rmdir: Try rmdir(/mnt/fs/TestDir)
fail_mkdir: Try mkdir(/mnt/fs/TestDir/WrTest1.txt)
succeed_unlink: Try unlink(/mnt/fs/TestDir/WrTest1.txt)
dev/
  console
  null
  ram0
mnt/
  fs/
    TESTDIR/
fail_stat: Try stat(/mnt/fs/TestDir/WrTest1.txt)
fail_mkdir: Try mkdir(/mnt/fs/TestDir)
succeed_rmdir: Try rmdir(/mnt/fs/TestDir)
dev/
  console
  null
  ram0
mnt/
  fs/
fail_stat: Try stat(/mnt/fs/TestDir)
succeed_mkdir: Try mkdir(/mnt/fs/NewDir1)
dev/
  console
  null
  ram0
mnt/
  fs/
    NEWDIR1/
succeed_stat: Try stat(/mnt/fs/NewDir1)
succeed_stat: stat(/mnt/fs/NewDir1) succeeded
/mnt/fs/NewDir1 stat:
	mode        : 000041b6
	type        : Directory
	size        : 0 (bytes)
	block size  : 512 (bytes)
	size        : 0 (blocks)
	access time : 0
	modify time : 0
	change time : 0
show_statfs: Try statfs(/mnt/fs/NewDir1)
show_statfs: statfs(/mnt/fs/NewDir1) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1972
	Free user blocks  : 1972
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
fail_mkdir: Try mkdir(/mnt/fs/NewDir1)
fail_stat: Try stat(/mnt/fs/NewDir1/WrTest2.txt)
write_test_file: opening /mnt/fs/NewDir1/WrTest2.txt for writing
write_test_file: wrote 20 bytes to /mnt/fs/NewDir1/WrTest2.txt
dev/
  console
  null
  ram0
mnt/
  fs/
    NEWDIR1/
      WRTEST2.TXT
succeed_stat: Try stat(/mnt/fs/NewDir1/WrTest2.txt)
succeed_stat: stat(/mnt/fs/NewDir1/WrTest2.txt) succeeded
/mnt/fs/NewDir1/WrTest2.txt stat:
	mode        : 000081b6
	type        : File
	size        : 20 (bytes)
	block size  : 512 (bytes)
	size        : 1 (blocks)
	access time : 0
	modify time : 0
	change time : 0
show_statfs: Try statfs(/mnt/fs/NewDir1/WrTest2.txt)
show_statfs: statfs(/mnt/fs/NewDir1/WrTest2.txt) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1971
	Free user blocks  : 1971
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
read_test_file: opening /mnt/fs/NewDir1/WrTest2.txt for reading
read_test_file: Read "This is a write test" from /mnt/fs/NewDir1/WrTest2.txt
fail_stat: Try stat(/mnt/fs/NewDir2)
succeed_mkdir: Try mkdir(/mnt/fs/NewDir2)
dev/
  console
  null
  ram0
mnt/
  fs/
    NEWDIR1/
      WRTEST2.TXT
    NEWDIR2/
succeed_stat: Try stat(/mnt/fs/NewDir2)
succeed_stat: stat(/mnt/fs/NewDir2) succeeded
/mnt/fs/NewDir2 stat:
	mode        : 000041b6
	type        : Directory
	size        : 0 (bytes)
	block size  : 512 (bytes)
	size        : 0 (blocks)
	access time : 0
	modify time : 0
	change time : 0
show_statfs: Try statfs(/mnt/fs/NewDir2)
show_statfs: statfs(/mnt/fs/NewDir2) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1970
	Free user blocks  : 1970
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
fail_rename: Try rename(/mnt/fs->/mnt/fs/NewDir3)
fail_rename: ERROR rename(/mnt/fs->/mnt/fs/NewDir3) succeeded
fail_rename: Try rename(/mnt/fs/NewDir1->/mnt/fs/NewDir2)
fail_rename: ERROR rename(/mnt/fs/NewDir1->/mnt/fs/NewDir2) succeeded
fail_stat: Try stat(/mnt/fs/NewDir3)
succeed_rename: Try rename(/mnt/fs/NewDir2->/mnt/fs/NewDir3)
dev/
  console
  null
  ram0
mnt/
  fs/
    NEWDIR3/
      NEWDIR1/
        WRTEST2.TXT
fail_stat: Try stat(/mnt/fs/NewDir2)
succeed_stat: Try stat(/mnt/fs/NewDir3)
succeed_stat: stat(/mnt/fs/NewDir3) succeeded
/mnt/fs/NewDir3 stat:
	mode        : 000041b6
	type        : Directory
	size        : 0 (bytes)
	block size  : 512 (bytes)
	size        : 0 (blocks)
	access time : 0
	modify time : 0
	change time : 0
show_statfs: Try statfs(/mnt/fs/NewDir3)
show_statfs: statfs(/mnt/fs/NewDir3) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1970
	Free user blocks  : 1970
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
fail_stat: Try stat(/mnt/fs/NewDir3/Renamed.txt)
succeed_rename: Try rename(/mnt/fs/NewDir1/WrTest2.txt->/mnt/fs/NewDir3/Renamed.txt)
succeed_rename: ERROR rename(/mnt/fs/NewDir1/WrTest2.txt->/mnt/fs/NewDir3/Renamed.txt) failed with errno=20
dev/
  console
  null
  ram0
mnt/
  fs/
    NEWDIR3/
      NEWDIR1/
        WRTEST2.TXT
fail_stat: Try stat(/mnt/fs/NewDir1/WrTest2.txt)
fail_stat: ERROR stat(/mnt/fs/NewDir1/WrTest2.txt) failed with errno=20(expected 2)
succeed_stat: Try stat(/mnt/fs/NewDir3/Renamed.txt)
succeed_stat: ERROR stat(/mnt/fs/NewDir3/Renamed.txt) failed with errno=2
show_statfs: Try statfs(/mnt/fs/NewDir3/Renamed.txt)
show_statfs: statfs(/mnt/fs/NewDir3/Renamed.txt) succeeded
	FS Type           : 4d44
	Block size        : 512
	Number of blocks  : 1984
	Free blocks       : 1970
	Free user blocks  : 1970
	Number file nodes : 0
	Free file nodes   : 0
	File name length  : 12
read_test_file: opening /mnt/fs/NewDir3/Renamed.txt for reading
read_test_file: ERROR failed to open /mnt/fs/NewDir3/Renamed.txt, errno=2
mount_main: Try unmount(/mnt/fs)
mount_main: 6 errors reported
```

**Comment by btashton** [comment]
This test was failing all the way back to 9.0 (maybe earlier).

**Comment by btashton** [comment]
This is resolved on the 10.0 branch with https://github.com/apache/incubator-nuttx-apps/pull/465
